assigment 4

orlando156 opened the pull request.

orlando156 committed:
assigment 4

orlando156 committed:
actualizacion de losejercicios

orlando156 committed:
actualizacion de los registros

orlando156 committed:
actualizacion de los ejercicios

se agregan los query de los ejercicios faltantes, dejando solo pendiente el ejemplo 2 y 3 del apartado de funciones

orlando156 committed:
actualizacion de los ejercicio 2 y 3

complemento los ejercicios 2 y 3 del apartado de funciones

orlando156 committed:
actualizacion del ejercicio 3 

se actualiza el ejercicio 3 del apartado de funciones, ya que el query anterior, funciona , pero no muestra el resultado deseado

rovalcor commented on assigment 4.sql:
```diff
@@ -0,0 +1,186 @@
+  SALARY
+  from Employees
+  WHERE 5 = ( SELECT COUNT(DISTINCT SALARY) FROM Employees AS em2 
+            WHERE em2.SALARY >= Employees.SALARY ); 
```
Bien! Pero la pregunta pide solo conseguir el salario. Solo para que lo tengas en mente

orlando156 replied:
query corregido

rovalcor commented on assigment 4.sql:
```diff
@@ -0,0 +1,186 @@
+  WHERE 5 = ( SELECT COUNT(DISTINCT SALARY) FROM Employees AS em2 
+            WHERE em2.SALARY >= Employees.SALARY ); 
+3) Write a query to find the 4th minimum salary in the employees table. 
+ select 
```
Lo mismo en este query.

orlando156 replied:
query corregido

rovalcor commented on assigment 4.sql:
```diff
@@ -0,0 +1,186 @@
+  Employees.EMPLOYEE_ID DESC
+LIMIT 10 
+5) Write a query to get the 3 maximum salaries. 
+SELECT
```
Igual aquí.

orlando156 replied:
query corregido

rovalcor commented on assigment 4.sql:
```diff
@@ -0,0 +1,186 @@
+  salary > (  SELECT MAX(salary)  FROM Employees  WHERE JOB_ID = 'SH_CLERK')
+  ORDER BY SALARY DESC
+2) Write a query to find the name (FIRST_NAME, LAST_NAME) of the employees who are not managers.
+ select FIRST_Name, LAST_NAME, 
```
Dos comentarios de este query:
1. No funciona correctamente porque no regresa ningún valor. El principal culpable es un MANAGER_ID que es NULL en tu subquery. [MySQL trata a los NULLs como un valor desconocido que regresa un NULL para cualquier operación matemática](http://www.sqlservercentral.com/articles/Advanced+Querying/fourrulesfornulls/1915/). Por ende, al comparar una lista con un NULL, ningún empleado va a hacer match. El secreto está en quitar ese NULL de lo que te regrese el subquery.
2. Usualmente es mala práctica hacer queries y publicarlos con comentarios en medio de cada línea.

rovalcor commented on assigment 4.sql:
```diff
@@ -0,0 +1,186 @@
+WHERE EMPLOYEE_ID NOT IN (SELECT Employees.MANAGER_ID FROM Employees) 
+
+3) Write a query to display the employee ID, first name, last name, and department names of all employees.
+SELECT Employees.EMPLOYEE_ID,
```
Whoops! Este ejercicio se me brincó de la parte de joins. Pero está bien 👍

rovalcor commented on assigment 4.sql:
```diff
@@ -0,0 +1,186 @@
+   LAST_NAME, 
+   salary
+	FROM Employees
+	WHERE salary > (SELECT  AVG(salary) FROM Employees)  
```
Aguas. La pregunta pide que se compare contra el salario promedio del departamento de ese empleado en particular.

rovalcor commented on assigment 4.sql:
```diff
@@ -0,0 +1,186 @@
+
+-- join queries*--
+1) Write a query to get the department name and number of employees in the department.
+SELECT DEPARTMENT_NAME,
```
Ojo a la pregunta: pide que saques el número de empleados en cada departamento.

orlando156 replied:
query corregido

rovalcor commented on assigment 4.sql:
```diff
@@ -0,0 +1,186 @@
+ON Jobs.JOB_ID = Employees.JOB_ID
+GROUP by EMPLOYEE_ID
+6) Write a query to display job title, employee name, and the difference between salary of the employee and minimum salary for the job.
+SELECT JOB_title,
```
Creo que hay que ver este query a detalle. No funciona por varios aspectos. Te recomiendo empieces de 0 con él y leas sobre el [uso de las aggregate functions](http://www.mysqltutorial.org/mysql-aggregate-functions.aspx) para que te des una idea de dónde vienes con el query y a dónde te interesa llegar con él.

rovalcor commented on assigment 4.sql:
```diff
@@ -0,0 +1,186 @@
+ORDER BY `Employees`.`EMPLOYEE_ID` ASC
+
+3) Write a query to display the department ID and name and first name of manager.
+SELECT Departments.DEPARTMENT_ID,
```
Disculpa si la pregunta no quedó clara. Lo que pide es el departamento y el nombre de sus managers.

orlando156 committed:
actualizacion del ejercicio 1

orlando156 committed:
actualizacion de los comentarios y ejemplos 

se actualiza los querys donde solo pide el salario y yo puse FIRST_NAME y LAST_NAME

también se actualiza ejercicio 1 de la parte de los joins

pendiente ejercicio 4 sunquerys 
ejercicio 3 de los join
ejercicio 6  de los join
